src/snek5000/util/restart.py

```diff
@@ -262,21 +269,182 @@ def load_for_restart(
+                "restart files. If not specified it would default to the "
+                "`path_session` value last recorded in the `params_simul.xml` file."
+            ),
+        )
+        parser.add_argument(
+            "--skip-verify-contents",
+            action="store_true",
+            help="Do not verify directory contents to avoid runtime errors.",
+        )
+        parser.add_argument(
+            "--add-to-end-time",
+            type=float,
+            default=None,
+            help="Time added to params.nek.general.end_time",
+        )
+        parser.add_argument(
+            "--end-time",
+            type=float,
+            default=None,
+            help="params.nek.general.end_time",
+        )
+        parser.add_argument(
+            "--num-steps",
+            type=int,
+            default=None,
+            help="params.nek.general.num_steps",
+        )
+
+        return parser
+
+    _str_command_after_simul = dedent(
+        """
+        # To visualize with IPython:
+
+        cd {path_run}; snek-ipy-load
+    """
+    )
+
+    def _get_params_simul_class(self, args):
+        if args.use_start_from is None and args.use_checkpoint is None:
+            logger.error("Either --use-start-from or --use-checkpoint have to be given")
+            sys.exit(1)
+        return load_for_restart(
+            args.path,
+            use_start_from=args.use_start_from,
+            use_checkpoint=args.use_checkpoint,
+            session_id=args.session_id,
+            verify_contents=not args.skip_verify_contents,
+            new_dir_results=args.new_dir_results,
+            only_check=args.only_check,
+        )
+
+    def _set_params_time_stepping(self, params, args):
+        if args.num_steps is not None:
+            params.nek.general.stop_at = "numSteps"
+            params.nek.general.num_steps = int(args.num_steps)
+        elif args.end_time is not None or args.add_to_end_time is not None:
+            params.nek.general.stop_at = "endTime"
+            if args.end_time is not None:
+                end_time = args.end_time
+            else:
+                end_time = float(params.nek.general.end_time) + args.add_to_end_time
+            params.nek.general.end_time = end_time
+
+    def _start_sim(self, sim, args):
+        sim.make.exec("run_fg", nproc=args.nb_mpi_procs)
+
+    def _check_params_time_stepping(self, params, path_file, args):
+        args_times = [args.num_steps, args.end_time, args.add_to_end_time]
+        if sum(arg is not None for arg in args_times) > 1:
+            raise ValueError(
+                "--add-to-end-time, --end-time and --num-steps are exclusive options."
+            )
+
+        if params.nek.general.stop_at == "numSteps" and args.num_steps is None:
+            raise ValueError(
+                'When params.nek.general.stop_at == "numSteps", '
+                "--num-steps should be given."
+            )
```

Comment:
Is this check necessary? We can override how the simulation stops in the restart run, can't we?

Anyways we are ensuring above:

```py
        if args.num_steps is not None:
            params.nek.general.stop_at = "numSteps"
```

Reply:
i removed the check which was not very useful and potentially problematic.